### PR TITLE
[mypy] Type annotations for graphs directory

### PR DESCRIPTION
**Pull request by Rohanrbharadwaj**
### Describe your change:

Related to #4052

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by cclauss — CHANGES_REQUESTED** [review]

File: graphs/breadth_first_search.py [hunk 1]
```diff
@@ -53,7 +53,7 @@ def bfs(self, start_vertex: int) -> set[int]:
         visited = set()
 
         # create a first in first out queue to store all the vertices for BFS
-        queue = Queue()
+        queue: Queue = Queue()
```
Comment (cclauss) [inline graphs/breadth_first_search.py:56]: Was mypy really failing this file just because of this annotation?!?

Comment (Rohanrbharadwaj) [reply graphs/breadth_first_search.py:56]: It was, I was surprised too.

File: mypy.ini [hunk 2]
```diff
@@ -2,4 +2,4 @@
 ignore_missing_imports = True
 install_types = True
 non_interactive = True
-exclude = (graphs/breadth_first_search.py|graphs/breadth_first_search_2.py|graphs/check_cycle.py|graphs/greedy_min_vertex_cover.py|matrix_operation.py|other/least_recently_used.py|other/lfu_cache.py|other/lru_cache.py|searches/simulated_annealing.py|searches/ternary_search.py)
+exclude = (graphs/check_cycle.py|graphs/greedy_min_vertex_cover.py|matrix_operation.py|other/least_recently_used.py|other/lfu_cache.py|other/lru_cache.py|searches/simulated_annealing.py|searches/ternary_search.py)
```
[the remark this answers is not in the document]
Comment (Rohanrbharadwaj) [reply mypy.ini:5]: On it

**Review by cclauss — APPROVED** [review]
Awesome.  The failing test is unrelated.

**Comment by Rohanrbharadwaj** [comment]
Should I do something about that (`web_programming.get_imdbtop.get_imdb_top_movies`) ??

**Comment by cclauss** [comment]
Web endpoints go down from time to time.  It is quite difficult to test those algorithms so we need to be a bit more tolerant of false fails.

**Pull request merged**